fooof/synth.py

```diff
@@ -17,6 +19,69 @@
 
 SynParams = namedtuple('SynParams', ['background_params', 'gaussian_params', 'nlv'])
 
+def param_iter(params):
+    """
+    This generator finds which parameter will be iterated through then
+    genertaes those iterations. params is a list given either as 
+    [cf, amp, bw] or [offset, slope]. The intended parameter to iterate
+    over will be replaced with a list in the form of [start, stop, step].
+    
+    Parameter
+    ---------
+    params : list
+        each element is a float except the parameter to be iterated over
+        that is a list itself
+        
+    Yields
+    -----
+        A 2 element tuple where the first value is a list of the next iterated 
+        parameters and the second element is the length of the generator
+
+    Example
+    -------
+    Example: if we want to iterate over center frequency values from
+    8 to 12 in increments of .25: params = [[8,12,.25], amp, bw]
+    """
+
+    
+    num_iters = 0
+    
+    for i in range(len(params)):
+        if isinstance(params[i], np.ndarray):
+            length = len(params[i])
+            num_iters += 1
+            ind = i
+    
+        if num_iters > 1:
+            raise Warning("Iteration is only supported on one parameter")
+            
+    temp = params
+    c = ( _param_generator(params,ind,x) for x in params[ind])     
+    return (c, length)
```

Comment:
This return doesn't need to return a tuple, right? It could be: return c, length?

Reply:
isnt the line:
return c, length 
an implicit tuple? 

https://www.geeksforgeeks.org/g-fact-41-multiple-return-values-in-python/